### ISSUE TITLE
networks.ingress value Additional properties are not allowed ('name' was unexpected)

### ISSUE THREAD

**Issue by lpeabody** [lmakarov/docksal-prod-poc#5]
Hey @lmakarov, thanks for putting together this sample of what a production setup might look like leveraging traefik for a reverse proxy alongside classic Docksal project images. This is more or less the setup I'm chasing (i.e. hoping to reuse Docksal's classic service offerings cli, web, db, varnish, etc.).

I'm running into a couple errors getting the system portion of this working. 

```
lpeabody@peabody-server01:~/server_configuration/system$ make up
Starting up containers for system...
#@sudo -E /usr/bin/docker-compose pull
ERROR: The Compose file './docker-compose.yml' is invalid because:
networks.ingress value Additional properties are not allowed ('name' was unexpected)
services.portainer.labels.traefik.enable contains true, which is an invalid type, it should be a string, number, or a null
services.traefik.labels.traefik.enable contains true, which is an invalid type, it should be a string, number, or a null
Makefile:27: recipe for target 'start' failed
make: *** [start] Error 1
```

My setup is a straight clone of the system setup you've documented on https://github.com/lmakarov/docksal-prod-poc/tree/master/system. I have .env with the following:

```
HOSTING_ENVIRONMENT = prod
```

And my .prod.env has the following:

```
# Settings for running docker/docker-compose
DOCKER ?= sudo -E /usr/bin/docker
DOCKER_COMPOSE ?= sudo -E /usr/bin/docker-compose

HTTP_PORT_MAPPING = 80:80

# Basic auth
# Generate a new credentials string with:
# echo $(htpasswd -nb '<AUTH-USER>' '<AUTH-PASS>') | sed -e 's/\$/\$\$/g'
# Credentials: admin:Ch@ngeMePle@se
ADMIN_BASIC_AUTH = userscrubbed:passscrubbed/
```

I googled a bit and found that this error:

```
networks.ingress value Additional properties are not allowed ('name' was unexpected)
```

Is due to network name being supported in compose file version 3.5+ (https://docs.docker.com/compose/compose-file/compose-versioning/#version-35), but the compose file version in system's docker-compose.yml is 2.1. I have not traced the other boolean-related errors yet. Just wondering if you could shine some light on this.

**Comment by lpeabody** [comment]
Looking at some Traefik docs, it looks like the labels should just be strings (https://docs.traefik.io/user-guide/examples/). I converted them accordingly. Still running into the error re: name not being a valid property of a network.

**Comment by lpeabody** [comment]
I got it working by updating to Docker Compose 1.24 and updating the version in system's docker-compose.yml file to 3.7.

**Comment by lmakarov** [comment]
@lpeabody that's weird. I just tried it locally and system services are starting just fine:

```
$ make start
Starting up containers for system...
#@docker-compose pull
Creating network "ingress" with the default driver
Creating portainer ... done
Creating traefik   ... done

$ make remove
Removing resources for system...
Stopping traefik   ... done
Stopping portainer ... done
Removing traefik   ... done
Removing portainer ... done
Removing network ingress
```

```
$ docker --version
Docker version 18.09.2, build 6247962
$ docker-compose --version
docker-compose version 1.23.2, build 1110ad01
```

```
$ docker version
Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:39 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:13:06 2019
  OS/Arch:          linux/amd64
  Experimental:     false

```

**Comment by lpeabody** [comment]
So your docker-compose.yml system file has version 2.1 specified but it's allowing the name property on the network?

**Comment by lpeabody** [comment]
So a couple details I left out:

I upgraded to docker-compose 1.24 from 1.17. When I ran `make up` with 1.17 it complained about name being unsupported in networking. I looked it up and name was introduced in Compose file version 3.5. So I upgraded to the latest Docker Compose, AND also changed the Compose version of system's docker-compose.yml to 3.7 (because why not use the latest).

That worked, but I never checked if leaving the version at 2.1 would work because I thought the error was telling me that 2.1 did not support the name property on networks, lol. Apparently it does? I'm confused though, if it was introduced in 3.5, then why would 2.1 build without error?

**Comment by lmakarov** [comment]
> So your docker-compose.yml system file has version 2.1 specified but it's allowing the name property on the network?

Yep.

I've just cloned the repo clean and verified it works. I see the docs stating the `name` property is not available until composer file format `3.5`, but it works fine in `2.1`.

The only issues is with the spaces in the `.env` files when using docker-compose v1.24. Apparently, it is more strict with the file format.

```
$ ./docker-compose --version
docker-compose version 1.24.0, build 0aa59064

$ DOCKER_COMPOSE=$(pwd)/docker-compose make config
/Users/leonid/Work/Labs/docksal/docksal-prod-poc/system/docker-compose config
ERROR: In file ./.env: environment variable name 'HOSTING_ENVIRONMENT ' may not contains whitespace.
make: *** [config] Error 1
```

Actually, compose loading that the `.env` file is unintentional, maybe even undesirable. The file is meant to be loaded by Makefile and not docker-compose.

**Comment by lpeabody** [comment]
Interesting! I guess it just probably auto-detects if it can support the content of the compose file, and if it can then it just does it... or maybe it's a bug :)

I'm going to stick pretty close to this setup, at least for the initial go of setting up this project's dev/stage/prod environments. Thank you for putting it together.
